src/main/resources/static/javascript/mainApp/timeout.controller.js

```diff
@@ -1,195 +1,117 @@
-     * @param $window - A reference to the browser's window object
-     * @param $uibModal - Creates modal
-     * @param dataProvider - Used for HTTP requests
-     * @param BASE_URL - Base url for api calls
-     * @param $timeout - AngularJS wrapper for window.setTimeout, used to implement session timeout
-     * @param $interval - AngularJS wrapper for window.setInterval, used to implement session timeout
+     * @param $scope - binding between controller and HTML page
+     * @param $window -
+     * @param $uibModal - creates modal
+     * @param dataProvider - http request
+     * @param BASE_URL - base url for api calls
      * @constructor
      */
-    function TimeoutJsController($scope, $window, $uibModal, $controller, dataProvider, BASE_URL, $timeout, $interval) {
+    function TimeoutJsController($scope, $window, $uibModal, $controller, dataProvider, BASE_URL) {
 
         angular.extend(this, $controller("GeneralJsController", { $scope: $scope }));
 
-        let createTimeoutModalPromise = {};
-        let countdownTimerPromise = {};
-        let isModalOpen = false;
-        const MAX_TIME_IDLE = 1000 * 60 * 25; // Time until modal opens - 25 minutes in milliseconds
-        const TIME_TO_LOGOUT = 60 * 5; // When modal opens, seconds left until user is logged out
-
-        $scope.secondsRemaining = TIME_TO_LOGOUT; // Seconds remaining before log out
-        $scope.timeRemaining = ""; // Formatted string of seconds remaining
-
-        angular.element(function () {
-            // Start timeouts
-            createTimeoutModalPromise = $timeout(() => {
-                $scope.createTimeoutModal();
-            }, MAX_TIME_IDLE);
-
-            // Attach callback when the createTimeoutModalPromise is canceled
-            createTimeoutModalPromise.then(() => {
-                // Timeout ends and function is executed
-            }, () => {
-                // User resets timer or function execution failed
-                restartTimeouts();
-            });
-
-            // If user clicks, reset timeout
+        $scope.seconds = 300;
+        $scope.idleTime = 0;
+        /**
+         * Every minute, checks whether or not user has clicked or pressed button
+         */
+        $(document).ready(function () {
+            //Increment the idle time counter every minute.
+            let idleInterval = setInterval($scope.timerIncrement, 60000); // 1 minute
+            //Zero the idle timer on mouse movement.
             $(this).click(function (e) {
-                if (!isModalOpen) {
-                    $timeout.cancel(createTimeoutModalPromise);
-                }
+                $scope.idleTime = 0;
             });
-            // If user presses a key on the keyboard reset timeout
+
             $(this).keypress(function (e) {
-                if (!isModalOpen) {
-                    $timeout.cancel(createTimeoutModalPromise);
-                }
+                $scope.idleTime = 0;
```

Comment:
Could this Jquery be switched out for angularJS functions? Let me know if there are any issues when using the angularJS equivalent functions.

Reply:
Disregard this